src/test/scala/medeia/generic/GenericCodecProperties.scala

```diff
@@ -54,4 +54,33 @@ class GenericCodecProperties extends Properties("GenericEncoding") {
     }
   }
 
+  //prevents unused field warnings
+  object ForHierarchyWithOptionsTest {
+    sealed trait Trait
+    case class A(stringField: String) extends Trait
+    case class B(int: Int) extends Trait
+
+    implicit val coproductDerivationOptions: CoproductDerivationOptions[Trait] =
+      CoproductDerivationOptions(typeNameTransformation = { case a => a.toLowerCase() }, typeNameKey = "otherType")
+    implicit val genericDerivationOptionsA: GenericDerivationOptions[A] = GenericDerivationOptions { case a => a.toLowerCase() }
+    implicit val genericDerivationOptionsB: GenericDerivationOptions[B] = GenericDerivationOptions { case a => a.toUpperCase() }
```

Comment:
Hmm that is a bit awkward, that we have to define the options for all subtypes as well...  Would be nice to only specify the options for the `Trait` and then it should apply to all subcases, otherwise it's easy to mess it up

Reply:
See #75  that would be possible, then the options for the trait are always more specific than the options for any subtypes, which means you are no longer able to override it, but I think that is the correct behavior and it would actually be a bug to have a mismatch.  WDYT?

Reply:
That would only be a solution if the Generic Encoders for case classes and sealed traits would use the same Option types (which they currently don't).

The thing is that both option types have a function (String => String) in them but the functions are used for different things.
Assuming that we provide String.toUpperCase for both:

In Coproducts the function specifies that `case class MyCaseClass(myField: String) extends SomeSealedTrait` is encoded as

```
{
  "type": "MYCASECLASS"
}
```
For case classes it specifies that `case class MyCaseClass(myField: String) ` is encoded as
```
{
  "MYFIELD" : "somevalue"
}
```

Therefore I opted to keep them separated from each other to not be forced to specify options for some feature I don't even use.

Maybe it would be better to merge the two option classes to one and make it contravariant? The downside would be that you could configure options for a case class that only make sense, when a sealed trait is derived (and vice versa if you write case class codecs from hand and only want the sealed trait logic to be done by medeia)

Reply:
I was not thinking about merging the two options type classes, as you said they have a different purpose :)

Making the Options for the `Product`Derivation contravariant is not what we want.

However for the Coproduct I think it's different, here having options for all subtypes of the sealed trait is error prone and code duplication.  

Anyway, seems like I got confused, the `val`s I deleted in #75  where vals for the product derivation and not coproduct :man_shrugging: